easybuild/tools/containers/docker.py

```diff
@@ -0,0 +1,153 @@
+from easybuild.tools.filetools import rmtree2
+from easybuild.tools.run import run_cmd
+
+
+DOCKER_TMPL_HEADER = """\
+FROM %(container_base)s
+LABEL maintainer=easybuild@lists.ugent.be
+"""
+
+DOCKER_INSTALL_EASYBUILD = """\
+RUN pip install -U pip setuptools && \\
+    hash -r pip && \\
+    pip install -U easybuild
+
+RUN mkdir /app && \\
+    mkdir /scratch && \\
+    mkdir /scratch/tmp && \\
+    useradd -m -s /bin/bash easybuild && \\
+    chown easybuild:easybuild -R /app && \\
+    chown easybuild:easybuild -R /scratch
+"""
+
+DOCKER_TMPL_FOOTER = """\
+USER easybuild
+
+RUN set -x && \\
+    . /usr/share/lmod/lmod/init/sh && \\
+    eb %(eb_opts)s --installpath=/app/ --prefix=/scratch --tmpdir=/scratch/tmp
+
+RUN touch ${HOME}/.profile && \\
+    echo '\\n# Added by easybuild docker packaging' >> ${HOME}/.profile && \\
+    echo 'source /usr/share/lmod/lmod/init/bash' >> ${HOME}/.profile && \\
+    echo 'module use %(init_modulepath)s' >> ${HOME}/.profile && \\
+    echo 'module load %(mod_names)s' >> ${HOME}/.profile
+
+CMD ["/bin/bash", "-l"]
+"""
+
+DOCKER_UBUNTU1604_INSTALL_DEPS = """\
+RUN apt-get update && \\
+    apt-get install -y python python-pip lmod curl wget
+
+RUN OS_DEPS='%(os_deps)s' && \\
+    test -n "${OS_DEPS}" && \\
+    for dep in ${OS_DEPS}; do apt-get -qq install ${dep} || true; done
+"""
+
+DOCKER_CENTOS7_INSTALL_DEPS = """\
+RUN yum install -y epel-release && \\
+    yum install -y python python-pip Lmod curl wget git
+
+RUN OS_DEPS='%(os_deps)s' && \\
+    test -n "${OS_DEPS}" && \\
+    yum --skip-broken install -y "${OS_DEPS}" || true
+"""
+
+DOCKER_OS_INSTALL_DEPS_TMPLS = {
+    DOCKER_BASE_IMAGE_UBUNTU: DOCKER_UBUNTU1604_INSTALL_DEPS,
+    DOCKER_BASE_IMAGE_CENTOS: DOCKER_CENTOS7_INSTALL_DEPS,
+}
+
+
+class DockerContainer(ContainerGenerator):
+
+    TOOLS = {'docker': None, 'sudo': None}
+
+    RECIPE_FILE_NAME = 'Dockerfile'
+
+    def resolve_template(self):
+        return (2 * "\n").join([
+            DOCKER_TMPL_HEADER % {'container_base': self.container_base},
+            DOCKER_OS_INSTALL_DEPS_TMPLS[self.container_base],
+            DOCKER_INSTALL_EASYBUILD,
+            DOCKER_TMPL_FOOTER,
+        ])
+
+    def resolve_template_data(self):
+        os_deps = det_os_deps(self.easyconfigs)
+
+        module_naming_scheme = ActiveMNS()
+
+        ec = self.easyconfigs[-1]['ec']
+
+        init_modulepath = os.path.join("/app/modules/all", *module_naming_scheme.det_init_modulepaths(ec))
+
+        mod_names = [ec['ec'].full_mod_name for ec in self.easyconfigs]
+
+        eb_opts = [os.path.basename(ec['spec']) for ec in self.easyconfigs]
+
+        return {
+            'os_deps': ' '.join(os_deps),
+            'eb_opts': ' '.join(eb_opts),
+            'init_modulepath': init_modulepath,
+            'mod_names': ' '.join(mod_names),
+        }
+
+    def validate(self):
+        if self.container_base not in [DOCKER_BASE_IMAGE_UBUNTU, DOCKER_BASE_IMAGE_CENTOS]:
```

Comment:
@FooBarQuaxx does it make sense to use `DOCKER_OS_INSTALL_DEPS_TMPLS.keys()` here?